vermouth/tests/test_molecule.py

```diff
@@ -889,17 +889,36 @@ def attribute_dict(draw, min_size=0, max_size=None):
     return draw(st.dictionaries(keys, values, min_size=min_size, max_size=max_size))
 
 
+@st.composite
+def parameter_effectors(draw):
+    possible_effectors = [
+        vermouth.molecule.ParamDistance,
+        vermouth.molecule.ParamAngle,
+        vermouth.molecule.ParamDihedral,
+        vermouth.molecule.ParamDihedralPhase,
+    ]
+    effector = draw(st.sampled_from(possible_effectors))
+    n_keys = effector.n_keys_asked
+    keys = draw(st.lists(
+        st.text(min_size=1, max_size=4),
+        min_size=n_keys, max_size=n_keys,
```

Comment:
Depending on how Hypothesis deals with min/max size, it may be faster to just make a list of size x with drawn elements:
`[draw(st.text(...) for _ in range(n_keys)]`